README


This archive contains the distribution Dist-Zilla-Plugin-CheckSelfDependency,
version 0.010:

  Check if your distribution declares a dependency on itself

This software is copyright (c) 2013 by Karen Etheridge.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.029.



Changes
Revision history for Dist-Zilla-Plugin-CheckSelfDependency

0.010     2015-01-03 20:48:28Z
          - bump Module::Metdata prereq for $VERSION parsing (RT#101095)

0.009     2014-12-03 02:07:51Z
          - increase minimum required version of CPAN::Meta::Prereqs (broken
            in 0.008)

0.008     2014-11-15 08:34:49Z
          - drop prereq requirement on Dist::Zilla 5.x
          - check against the version being required, not just the module,
            when this is relevant (for develop prereqs)

0.007     2014-01-25 03:20:52Z
          - now using a filefinder to find all modules to check -- now only
            checking files in lib/ by default

0.006     2013-12-07 22:27:04Z
          - now also checking for develop prereqs, that aren't listed in
            'provides' metadata

0.005     2013-10-26 01:31:06Z
          - set minimum version prereq for Module::Metadata

0.004     2013-10-24 04:17:27Z
          - fixes for Dist::Zilla 5.x, where we need to set the binmode on the
            fh before opening to the encoded content (we are still compatible
            with older Dist::Zilla as well)

0.003     2013-10-18 01:13:36Z
          - properly handle .pm files that don't exist on disk yet (e.g.
            ::Conflicts)

0.002     2013-10-17 02:58:11Z
          - skip checking develop prereqs, (it is not horrible for a plugin or
            bundle to list itself as a develop prereq)

0.001     2013-10-14 02:15:52Z
          - Initial release.


Makefile.PL
# This Makefile.PL for Dist-Zilla-Plugin-CheckSelfDependency was generated by
# Dist::Zilla::Plugin::MakeMaker::Fallback 0.016
# and Dist::Zilla::Plugin::MakeMaker::Awesome 0.32.
# Don't edit it but the dist.ini and plugins used to construct it.

use strict;
use warnings;

BEGIN {
my %configure_requires = (
    'Module::Build::Tiny' => '0.039',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";
}
else
{
    $ENV{PERL_MM_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

If you're installing manually, please retrain your fingers to run Build.PL
when present instead.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are doing something overly clever, in which case you
should consider setting the 'prefer_installer' config option in CPAN.pm, or
'prefer_makefile' in CPANPLUS, to 'mb" and '0' respectively.

You can also silence this warning for future installations by setting the
PERL_MM_FALLBACK_SILENCE_WARNING environment variable.

EOW

    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));
}
}

use 5.008;
use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Check if your distribution declares a dependency on itself",
  "AUTHOR" => "Karen Etheridge <ether\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "Module::Build::Tiny" => "0.039"
  },
  "DISTNAME" => "Dist-Zilla-Plugin-CheckSelfDependency",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.008",
  "NAME" => "Dist::Zilla::Plugin::CheckSelfDependency",
  "PREREQ_PM" => {
    "CPAN::Meta::Prereqs" => "2.132830",
    "CPAN::Meta::Requirements" => 0,
    "Dist::Zilla::Plugin::MetaResources" => "4.300039",
    "Dist::Zilla::Role::AfterBuild" => 0,
    "Dist::Zilla::Role::FileFinderUser" => 0,
    "Module::Metadata" => "1.000023",
    "Moose" => 0,
    "namespace::autoclean" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Dist::Zilla::Role::MetaProvider" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Path::Tiny" => 0,
    "Test::DZil" => 0,
    "Test::Deep" => 0,
    "Test::Fatal" => 0,
    "Test::More" => 0,
    "if" => 0,
    "utf8" => 0
  },
  "VERSION" => "0.010",
  "test" => {
    "TESTS" => "t/*.t"
  }
);

my %FallbackPrereqs = (
  "CPAN::Meta::Prereqs" => "2.132830",
  "CPAN::Meta::Requirements" => 0,
  "Dist::Zilla::Plugin::MetaResources" => "4.300039",
  "Dist::Zilla::Role::AfterBuild" => 0,
  "Dist::Zilla::Role::FileFinderUser" => 0,
  "Dist::Zilla::Role::MetaProvider" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Module::Build::Tiny" => "0.039",
  "Module::Metadata" => "1.000023",
  "Moose" => 0,
  "Path::Tiny" => 0,
  "Test::DZil" => 0,
  "Test::Deep" => 0,
  "Test::Fatal" => 0,
  "Test::More" => 0,
  "if" => 0,
  "namespace::autoclean" => 0,
  "strict" => 0,
  "utf8" => 0,
  "warnings" => 0
);

unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);


Build.PL
# This Build.PL for Dist-Zilla-Plugin-CheckSelfDependency was generated by
# Dist::Zilla::Plugin::ModuleBuildTiny::Fallback 0.014
use strict;
use warnings;

my %configure_requires = (
    'Module::Build::Tiny' => '0.039',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";

    # This section for Dist-Zilla-Plugin-CheckSelfDependency was generated by Dist::Zilla::Plugin::ModuleBuildTiny 0.008.
    use strict;
    use warnings;

    use 5.008;
    # use Module::Build::Tiny 0.039;
    Module::Build::Tiny::Build_PL();
}
else
{
    $ENV{PERL_MB_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are running this file manually, in which case you need
to learn to first fulfill all configure requires prerequisites listed in
META.yml or META.json -- or use a cpan client to install this distribution.

You can also silence this warning for future installations by setting the
PERL_MB_FALLBACK_SILENCE_WARNING environment variable, but please don't do
that until you fix your toolchain as described above.

EOW
    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));


    # This section was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.029.
    use strict;
    use warnings;

    require Module::Build; Module::Build->VERSION(0.28);


    my %module_build_args = (
      "build_requires" => {},
      "configure_requires" => {
        "Module::Build::Tiny" => "0.039",
        "perl" => "5.006"
      },
      "dist_abstract" => "Check if your distribution declares a dependency on itself",
      "dist_author" => [
        "Karen Etheridge <ether\@cpan.org>"
      ],
      "dist_name" => "Dist-Zilla-Plugin-CheckSelfDependency",
      "dist_version" => "0.010",
      "license" => "perl",
      "module_name" => "Dist::Zilla::Plugin::CheckSelfDependency",
      "recommends" => {},
      "recursive_test_files" => 1,
      "requires" => {
        "CPAN::Meta::Prereqs" => "2.132830",
        "CPAN::Meta::Requirements" => 0,
        "Dist::Zilla::Plugin::MetaResources" => "4.300039",
        "Dist::Zilla::Role::AfterBuild" => 0,
        "Dist::Zilla::Role::FileFinderUser" => 0,
        "Module::Metadata" => "1.000023",
        "Moose" => 0,
        "namespace::autoclean" => 0,
        "perl" => "5.008",
        "strict" => 0,
        "warnings" => 0
      },
      "script_files" => [],
      "test_requires" => {
        "Dist::Zilla::Role::MetaProvider" => 0,
        "ExtUtils::MakeMaker" => 0,
        "File::Spec" => 0,
        "Path::Tiny" => 0,
        "Test::DZil" => 0,
        "Test::Deep" => 0,
        "Test::Fatal" => 0,
        "Test::More" => 0,
        "if" => 0,
        "perl" => "5.008",
        "utf8" => 0
      }
    );


    my %fallback_build_requires = (
      "Dist::Zilla::Role::MetaProvider" => 0,
      "ExtUtils::MakeMaker" => 0,
      "File::Spec" => 0,
      "Path::Tiny" => 0,
      "Test::DZil" => 0,
      "Test::Deep" => 0,
      "Test::Fatal" => 0,
      "Test::More" => 0,
      "if" => 0,
      "perl" => "5.008",
      "utf8" => 0
    );


    unless ( eval { Module::Build->VERSION(0.4004) } ) {
      delete $module_build_args{test_requires};
      $module_build_args{build_requires} = \%fallback_build_requires;
    }

    my $build = Module::Build->new(%module_build_args);

    $build->create_build_script;
}
